Src/CsvLINQPadDriver/Extensions/FileExtensions.cs

```diff
@@ -536,7 +537,7 @@ Encoding GetEncoding()
             {
                 Charset charset;
 
-                using (var stream = File.OpenRead(fileName))
+                using (var stream = new FileStream(fileName, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
```

Comment:
`FileShare.ReadWrite` means that other processes can read and write to a file opened by this process. It doesn't solve locked file issue, i.e. if another process opens file without specifying `FileShare.Read` nothing can be done about it.

`File.OpenRead` currently used internally creates file stream with `FileShare.Read`: `new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read)`

Reply:
I use NLog for writing tab separated files and unfortunately driver was not been able to open such files, I have daily rollover so current file can't be read. My idea was to add that possibility so I can perform queries in such files, I'm fine with the option as long as I can open such file.

Reply:
Of course if file is locked by application there is no help, but opening files like proposed helps when reading logger files.

Reply:
Implemented in [6.20.0](https://www.nuget.org/packages/CsvLINQPadDriver/6.20.0).

Check `Do not lock files being read option` to get expected behavior.